elements/lisk-chain/src/chain.ts

```diff
@@ -338,7 +346,12 @@ export class Chain {
 	}
 
 	public validateBlockHeader(block: Block): void {
-		validatePreviousBlockProperty(block, this.genesisBlock);
+		// Validate block header
+		const errors = validator.validate(blockHeaderSchema, block.header);
+		if (errors.length) {
+			throw new Error(errors[0].message);
```

Comment:
it is returning the error array, so i dont think this works

Reply:
ah you mean parse it to the error, but I think if it just return the particular error instead of asking it to handle here, it would be better.
but I do agree we should use the single error everywhere and we should fix it.

Reply:
Just for reference can you create an issue to remove all usages of throwing errors like `throw new Error(errors[0].message)`